portal/static/js/main.js

```diff
@@ -2112,66 +2112,103 @@ var assembleContent = {
 
             var studyIdField = $("#profileStudyId");
             var siteIdField = $("#profileSiteId");
-            var hasStudyId = studyIdField.length > 0 && studyIdField.is(":visible");
-            var hasSiteId = siteIdField.length > 0 && siteIdField.is(":visible");
+            /*
+             * siteId field is only present in Truenth
+             * studyId field is only present in Eproms
+             */
+            var hasStudyIdField = studyIdField.length > 0 && studyIdField.is(":visible");
+            var hasSiteIdField = siteIdField.length > 0 && siteIdField.is(":visible");
             var studyId = studyIdField.val();
             var siteId = siteIdField.val();
 
 
-            if (hasStudyId || hasSiteId) {
-                var identifiers = null;
-                //get current identifier(s)
-                $.ajax ({
-                    type: "GET",
-                    url: "/api/demographics/"+userId,
-                    async: false
-                }).done(function(data) {
-                    if (data && data.identifier) {
-                        identifiers = [];
-                        (data.identifier).forEach(function(identifier) {
-                            if (identifier.system != SYSTEM_IDENTIFIER_ENUM["external_study_id"] &&
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["external_site_id"] &&
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["practice_region"]) {
-                                identifiers.push(identifier);
-                            }
-                        });
-                    }
-                }).fail(function(xhr) {
-                   tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
-                });
+            var identifiers = null;
+            /*
+             * get current identifier(s)
+             */
+            $.ajax ({
+                type: "GET",
+                url: "/api/demographics/"+userId,
+                async: false
+            }).done(function(data) {
+                if (data && data.identifier) {
+                    identifiers = [];
+                    (data.identifier).forEach(function(identifier) {
+                        identifiers.push(identifier);
+                    });
+                }
+            }).fail(function(xhr) {
+               tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
+            });
 
-                if (hasStudyId) {
+            /*
+             * NOTE: this will save study Id or site Id only if each has a value
+             * otherwise if each is empty, it will be purged from the identifiers
+             */
+
+            if (hasStudyIdField || hasSiteIdField) {
```

Comment:
It might be less work to create an array and push `studyIdObj`, `siteIdObj` if `studyId` and `siteId` have good values, then merge that array to `identifiers`. 

I think with @pbugni changes to the demographics API you can always expect good identifiers (no empty values), so you shouldn't have to do any filtering.